pkg/operator/diff/internal/diff.go

```diff
@@ -186,14 +189,58 @@ func (g *DiffGenerator) Run(oldModel, newModel model.Model) (model.Model, error)
 		var outputHasDefault bool
 		outputPkg.DefaultChannel, outputHasDefault = outputPkg.Channels[newPkg.DefaultChannel.Name]
 		if !outputHasDefault {
-			// Create a name-only channel since oldModel contains the channel already.
-			outputPkg.DefaultChannel = copyChannelNoBundles(newPkg.DefaultChannel, outputPkg)
+			// Set the defaultChannel using the priority of a channel when the default got filtered out
+			// If no channels with the Priority property, raise an error
+			if err := setDefaultChannel(outputPkg); err != nil {
+				return outputModel, err
+			}
 		}
 	}
 
 	return outputModel, nil
 }
 
+type ChannelPriorityPropList []property.Channel
+
+// setDefaultChannel sets the new default channel of a package if the old default channel got filtered out.
+// Throws an error if there are no channels with the Priority property
+func setDefaultChannel(outputPkg *model.Package) error {
+	p := make(ChannelPriorityPropList, len(outputPkg.Channels))
+	i := 0
+	for _, channel := range outputPkg.Channels {
+		var channelPriority property.Channel
+		for _, prop := range channel.Properties {
+			if prop.Type == "olm.channel" {
+				json.Unmarshal(prop.Value, &channelPriority)
+				p[i] = channelPriority
+				i++
+			}
+		}
+	}
+
+	if i > 0 {
+		sort.Slice(p, func(j, k int) bool { return p[j].Priority < p[k].Priority })
+
+		klog.V(1).Infof("defaultChannel choices sorted by priority for package: %s\n", outputPkg.Name)
+		for _, k := range p {
+			klog.V(1).Infof("%v\t%v\n", k.ChannelName, k.Priority)
+		}
+
+		// pick last channel as it is the one with the highest priority
+		chosenChannelName := p[len(p)-1].ChannelName
+		for chname, channel := range outputPkg.Channels {
+			if chname == chosenChannelName {
+				outputPkg.DefaultChannel = channel
+				klog.V(0).Infof("Newly assigned default channel from existing channels by Priority is %s\n", chosenChannelName)
+			}
+		}
+		return nil
+	}
+
+	return fmt.Errorf("unable to set a default channel for package: %s - no remaining channels in the filtered output have the Priority property", outputPkg.Name)
```

Comment:
- variable `p` is too short… variables should be more descriptive
- variable `p` creates a resulting slice that is as long as your output channel, but it **_should not_** be. The size of the slice `p` is not determined solely by the number of channels. Each channel can have zero or more properties, and some properties may not be “olm.channel”. Lines 210 through 219 selectively add an item at the beginning of the slice, _potentially_ leaving “empty” channels at the end (e.g. if there were other properties that are not “olm.channel”, you could end up with some slice entries with `property.Channel {ChannelName: "", Priority: 0}`). On the other hand, if you have multiple “olm.channel” entires, you could assign an index at line 215 that exceeds the size of the slice. I assume your logging at lines 225 to 227 would potentially show an empty channel name with a zero priority for some of the items. 
  - I think at the very least you should start with an empty slice for variable `p` and only append values that match type “olm.channel” to avoid adding “invalid” channels in your slice. 
  - It may be useful to highlight in the log any channels that had a duplicate priority value since that's a sign that someone did something incorrectly.